src/views/Profile/PersonalInfo/PersonalInfo.tsx

```diff
@@ -200,6 +203,20 @@ export const PersonalInfo: React.FC<IEditAuthorProps> = ({ author }) => {
     [errorMessages],
   );
 
+  useEffect(() => {
+    const handleBeforeUnload = (e: BeforeUnloadEvent) => {
+      if (!_isEqual(previousAuthorValues, newAuthorValues)) {
+        e.preventDefault();
+        e.returnValue =
+          'Зміни не збережені. Ви впевнені, що хочете залишити сторінку?';
+      }
+    };
+    window.addEventListener('beforeunload', handleBeforeUnload);
+    return () => {
+      window.removeEventListener('beforeunload', handleBeforeUnload);
+    };
+  }, [newAuthorValues]);
```

Comment:
I feel anxious about not listing `previousAuthorValues` in the dependencies list. Also I'd expect ESLint to complain. May we add it there?

Also, do we have ESLint enabled in the project?

Reply:
I will add